src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/Analyzer.scala

```diff
@@ -0,0 +1,20 @@
+package com.microsoft.partnercatalyst.fortis.spark.analyzer
+
+import com.microsoft.partnercatalyst.fortis.spark.dto.{Location, Tag}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.image.ImageAnalyzer
+import com.microsoft.partnercatalyst.fortis.spark.transforms.language.LanguageDetector
+import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.LocationsExtractor
+import com.microsoft.partnercatalyst.fortis.spark.transforms.people.PeopleRecognizer
+import com.microsoft.partnercatalyst.fortis.spark.transforms.sentiment.SentimentDetector
+import com.microsoft.partnercatalyst.fortis.spark.transforms.topic.KeywordExtractor
+
+trait Analyzer[T] {
+  type LocationFetcher = (Double, Double) => Iterable[Location]
+
+  def toSchema(item: T, locationFetcher: LocationFetcher, imageAnalyzer: ImageAnalyzer): ExtendedDetails[T]
```

Comment:
Have I omitted this from the original implementation, or is this something that we still need to bring online with an additional changeset? I'd seen gender stuff in an earlier PR, but analysis wasn't yet being performed IIRC.